plaso/analysis/browser_search.py

```diff
@@ -308,26 +338,25 @@ def CompileReport(self, mediator):
     analysis_report.report_dict = results
     return analysis_report
 
-  def ExamineEvent(self, mediator, event):
+  def ExamineEvent(self, mediator, event, event_data):
     """Analyzes an event.
 
     Args:
       mediator (AnalysisMediator): mediates interactions between
           analysis plugins and other components, such as storage and dfvfs.
-      event (EventObject): event to examine.
+      event (EventObject): event.
+      event_data (EventData): event data.
     """
-    # This event requires an URL attribute.
-    url = getattr(event, 'url', None)
-    if not url:
+    if event_data.data_type not in self._SUPPORTED_EVENT_DATA_TYPES:
       return
 
-    # TODO: refactor this the source should be used in formatting only.
-    # Check if we are dealing with a web history event.
-    source, _ = formatters_manager.FormattersManager.GetSourceStrings(event)
-
-    if source != 'WEBHIST':
+    url = getattr(event_data, 'url', None)
+    if not url:
       return
 
+    parser_or_plugin_name = getattr(event_data, 'parser', 'N/A')
```

Comment:
parser is set on event, not event_data: https://github.com/log2timeline/plaso/blob/master/plaso/parsers/mediator.py#L412

Reply:
After looking at this a bit more, I think this is actually set on the event_data after all: https://github.com/log2timeline/plaso/blob/master/plaso/parsers/mediator.py#L503 but it's quite confusing from the method naming etc.

Also, I'm not sure whether "parser" should be set on the event_data or the event or both. It doesn't seem to be defined on either though: https://github.com/log2timeline/plaso/blob/master/plaso/containers/events.py and it should be defined _somewhere_.

Reply:
> but it's quite confusing from the method naming etc.

Agree that's why I'm cleaning it up

Reply:
> I'm not sure whether "parser" should be set on the event_data 

It is https://github.com/log2timeline/plaso/blob/master/plaso/parsers/mediator.py#L412

This might be a bit hard to follow (and needs cleanup, hence this PR) 

But it is passed event data from https://github.com/log2timeline/plaso/blob/master/plaso/parsers/mediator.py#L503